Makefile.am

```diff
@@ -8,7 +8,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 AM_CFLAGS = -Wall \
-            -std=gnu99 \
+            -std=gnu11 \
```

Comment:
I think we should not change this right now (in the future, maybe), because **cglm** should work with C89 and C99. Upgrading C version may cause to ignore some warnings for old versions. We cannot suppress warnings by upgrading C version. 

Removing `-pedantic` would also suppress warnings if there is any about structs.

Reply:
It works just fine with C99, but you need to build it with `-DCGLM_NO_ANONYMOUS_STRUCT`. Anonymous structures are a C11 feature, so cglm is already using it. Pretending to build in C99 mode while using a C11 feature is strange.

A _user_ of the library can always use C99. They just `#define CGLM_NO_ANONYMOUS_STRUCT` before `#include`ing the CGLM headers. So even if we build with C11, we don't break any compatibility.

Maybe this can just be added to the README? I can add a section that explains all this and add it to this PR.

Please don't remove `-pedantic` though, that's just gonna push the warnings to the user.

Reply:
Okay,


>  Anonymous structures are a C11 feature, so cglm is already using it. Pretending to build in C99 mode while using a C11 feature is strange.


I think it can be used without C11, with language extensions, so "pretending" is not really true here. If compiler uses C89/C99 and doesn't have any extension about anonymous structs then user can define `CGLM_NO_ANONYMOUS_STRUCT` to disable it. 

But switching to C11 seems not to be bad idea. Let's do it.

Another option, maybe we could check compiler features, version... to enable/disable it automatically ?

> Maybe this can just be added to the README? I can add a section that explains all this and add it to this PR.

Please.

Contribution about README, Documentation, Tests are also welcome ;)

> Please don't remove -pedantic though, that's just gonna push the warnings to the user.

Let's keep it.

Reply:
> Another option, maybe we could check compiler features, version... to enable/disable it automatically ?

I can implement that. It involves a bit of a dance to stay backward-compatible and such, but it should be possible. I'll be amending the PR as such.